examples/codebert_example.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Guillaume Becquin
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use rust_bert::codebert::{
+    CodeBertConfigResources, CodeBertForFeatureExtractionConfig, CodeBertForFeatureExtractionModel,
+    CodeBertMergesResources, CodeBertModelResources, CodeBertVocabResources,
+};
+use rust_bert::pipelines::common::ModelType;
+use rust_bert::pipelines::masked_language::{MaskedLanguageConfig, MaskedLanguageModel};
+use rust_bert::pipelines::sequence_classification::{
+    SequenceClassificationConfig, SequenceClassificationModel,
+};
+use rust_bert::resources::{LocalResource, RemoteResource};
+use std::path::PathBuf;
+fn feature_extraction() -> Result<()> {
+    //    Set-up model
+    let config = CodeBertForFeatureExtractionConfig::new(
+        ModelType::CodeBert,
+        RemoteResource::from_pretrained(CodeBertModelResources::CODEBERT),
+        None,
+        RemoteResource::from_pretrained(CodeBertConfigResources::CODEBERT),
+        RemoteResource::from_pretrained(CodeBertVocabResources::CODEBERT),
+        RemoteResource::from_pretrained(CodeBertMergesResources::CODEBERT),
+        true,
+        None,
+        None,
+    );
+
+    let feature_extraction_model = CodeBertForFeatureExtractionModel::new(config)?;
+    //    Define input
+    let input = ["this is an example sentence", "each sentence is converted"];
+
+    //    Run model
+    let output = feature_extraction_model.predict(&input);
+    for hidden_states in output {
+        println!("{:?}", hidden_states);
+    }
+
+    Ok(())
+}
+
+fn masked_language() -> Result<()> {
+    let config = MaskedLanguageConfig::new(
+        ModelType::CodeBert,
+        RemoteResource::from_pretrained(CodeBertModelResources::CODEBERT_MLM),
+        // None,
+        LocalResource {
+            local_path: PathBuf::from("/home/vincent/.cache/model/codebert-base-mlm/rust_model.ot"),
+        },
+        RemoteResource::from_pretrained(CodeBertConfigResources::CODEBERT_MLM),
+        RemoteResource::from_pretrained(CodeBertVocabResources::CODEBERT_MLM),
+        Some(RemoteResource::from_pretrained(
+            CodeBertMergesResources::CODEBERT_MLM,
+        )),
+        true,
+        None,
+        false,
+    );
+    //    Set-up model
+    let model = MaskedLanguageModel::new(config)?;
+    //    Define input
+    let input = [
+        "Looks like one <mask> is missing!",
```

Comment:
> 

Hello@guillaume-be
Thanks for your review.
1. I tried to load local model as your suggestion, but error ocuurs: "the trait `ResourceProvider` is not implemented for `Box<LocalResource>`" and "expected struct `Box`, found struct `RemoteResource`", as described in following image.
![image](https://user-images.githubusercontent.com/9975892/190201855-b8c97588-238c-4816-aa3b-a76b2d5fb1f1.png)
![image](https://user-images.githubusercontent.com/9975892/190201783-4aa4d150-97fa-4a87-88fc-c68b77c65b11.png)

2. I will add an extra configuration option for pipelines::masked_language as you suggested.
3. The codebert has an identical  model to roberta. However, some models in huggingface.co has a different prefix as roberta. For example, the model structure of roberta always is like "roberta.encoder.layer....."; some codebert model in  huggingface.co is "encoder.layer"; some is "roberta.encoder.layer"; some is "codebert.encoder.layer". Error would occurs while loading codebert model like a roberta way.Hence, supporting codebert in a seperate way may be useful to tackle the prefix difference.

Reply:
Hello @guillaume-be Could you give me some advice on the first and third question? I am going to finish this PR as soon as possible!

Reply:
Hello @Vincent-Xiao ,

Regarding 1. the error occurs because you are trying to mix `LocalResource` with `RemoteResource`. While the models can take either of them (pointing to a remote model or local model files), it is not possible to mix both. In your example you would have to update the config and tokenizer files to also point to local files.

Rgarding 3., could you give an example of a model that uses a different prefix for the model (`codebert.encoder.layer`)? I'd be interested to see how the Python version fixes this problem as the Python's implementation of transformers does not include code specific to `codebert`

Reply:
Hello @guillaume-be 
1. My implemented method for loading local model in this PR supports mix LocalResource with RemoteResource. I think supporting this mix way still means something. The huggingface.co always have config files but may not contains a rust model. However, if you don't think it's worth supporting this, I will withdraw the sequenceConfig update.
3. The prefix of microsoft/codebert-base is "encoder.layer" without "roberta.".  And our team trained a codebert-based model (Vincent-Xiao/codebert-curs) to classify the "unsafe" keywords in rust program. This mode has a "codebert.encoder.layer" prefix.

Reply:
@guillaume-be 
1. There are an common scenario where huggingface.co/models don't have a rust_mode.ot of a specific bert . In such circumstance, if someone want to deploy this specific model in rust-bert, user must convert the pytorch_model.bin to rust_model.ot offline. And user may load local model files and remote config files (such as config.json or merges.txt from huggingface.co/models). Therefore, it's necessary to support the mixed way (pointing to a remote config and local model files). My implemented LanguageConfig support this scenario well.

3. I have withdrawn the codebert support in consideration of your suggestion.

Reply:
Hello @Vincent-Xiao ,

For 1. there is a way to implement such functionality without having to duplicate field. The issue is that the generic `<R>` is shared between all resource types. I am pushing a pull request hat relaxes this restriction, see https://github.com/guillaume-be/rust-bert/pull/291. Can you please update the style of your proposal to match the linked pull request? This should eliminate the need for a duplicate `local_model_resource`.

The following works on the proposed implementation:
```rust
extern crate anyhow;

use rust_bert::pipelines::common::ModelType;
use rust_bert::pipelines::sequence_classification::{
    SequenceClassificationConfig, SequenceClassificationModel,
};
use rust_bert::resources::{LocalResource, RemoteResource};
use tch::Device;

fn main() -> anyhow::Result<()> {
    //    Set-up model
    let config = SequenceClassificationConfig {
        model_type: ModelType::Roberta,
        model_resource: Box::new(LocalResource {
            local_path: "path/to/CodeBERTa-language-id/rust_model.ot".to_string(),
        }),
        config_resource: Box::new(RemoteResource {
            url:
                "https://huggingface.co/huggingface/CodeBERTa-language-id/resolve/main/config.json"
                    .to_string(),
            cache_subdir: "codeberta/config".to_string(),
        }),
        vocab_resource: Box::new(RemoteResource {
            url: "https://huggingface.co/huggingface/CodeBERTa-language-id/resolve/main/vocab.json"
                .to_string(),
            cache_subdir: "codeberta/vocab".to_string(),
        }),
        merges_resource: Some(Box::new(RemoteResource {
            url: "https://huggingface.co/huggingface/CodeBERTa-language-id/resolve/main/merges.txt"
                .to_string(),
            cache_subdir: "codeberta/merges".to_string(),
        })),
        lower_case: false,
        strip_accents: None,
        add_prefix_space: None,
        device: Device::cuda_if_available(),
    };

    let sequence_classification_model = SequenceClassificationModel::new(config)?;

    //    Define input
    let input = [
        "
        def f(x):
            return x**2
        ",
        "const foo = 'bar'",
        "outcome := rand.Intn(6) + 1",
    ];

    //    Run model
    let output = sequence_classification_model.predict(&input);
    for label in output {
        println!("{:?}", label);
    }

    Ok(())
}
```

Reply:
@guillaume-be Thanks for your suggestion. I have revoked this functiuon and updated the the implemnetation for local moddel support as your suggesting way.